libgnucash/app-utils/gnc-ui-util.c

```diff
@@ -1863,6 +1863,39 @@ xaccPrintAmount (gnc_numeric val, GNCPrintAmountInfo info)
     return buf;
 }
 
+const char *
+gnc_print_amount_with_bidi_ltr_isolate (gnc_numeric val, GNCPrintAmountInfo info)
+{
+    /* hack alert -- this is not thread safe ... */
+    static char buf[1024];
+
+    if (!xaccSPrintAmount (buf, val, info))
+        buf[0] = '\0';
+
+    if (buf[0] && info.use_symbol)
+    {
+        char temp_buf[1024];
+
+        strcpy (temp_buf, "\u2066"); // ltr isolate
+        strcat (temp_buf, buf);
+        strcat (temp_buf, "\u2069"); // pop directional formatting
```

Comment:
If you decide I've gone a bit over with a single buffer and for future reference it's faster to use `stpcpy`:
```
    char *pos = stpcpy (temp_buf, "\u2066");
    pos = stpncpy (pos, buf, buflen - 7);
    pos = stpcpy (pos, "\u2069");
```
Because it saves finding the end of the string every time.

Reply:
Used your suggestion.